### PR TITLE
Resolve issues detected by cppcheck

### PR DESCRIPTION
**Pull request by chipitsine**
[client/X11/xf_rail.c:205] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:206] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:207] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:208] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:215] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:216] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:217] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.
[client/X11/xf_rail.c:218] -> [client/X11/xf_rail.c:220]: (warning) Either the condition 'if(appWindow)' is redundant or there is possible null pointer dereference: appWindow.

### CODE REVIEW

**Comment by akallabeth** [comment]
@chipitsine Thank you for your work, but if you have other issues left, could you bundle them in a single pull?

**Comment by chipitsine** [comment]
yes, there are issues, they are "questionable", so, I guess some of them might be accepted, some rejected.

ok, I'll make a bundle

**Comment by akallabeth** [comment]
@freerdp-bot test

File: client/X11/xf_rail.c [hunk 1]
```diff
@@ -217,12 +217,9 @@ void xf_rail_invalidate_region(xfContext* xfc, REGION16* invalidRegion)
 				updateRect.right = extents->right - appWindow->x;
 				updateRect.bottom = extents->bottom - appWindow->y;
 
-				if (appWindow)
-				{
-					xf_UpdateWindowArea(xfc, appWindow, updateRect.left, updateRect.top,
-					                    updateRect.right - updateRect.left,
-					                    updateRect.bottom - updateRect.top);
-				}
+				xf_UpdateWindowArea(xfc, appWindow, updateRect.left, updateRect.top,
```
Comment (akallabeth) [inline client/X11/xf_rail.c]: Would suggest to remove this check for all usages of this function and add one before dereferencing the arguments within the function.

File: channels/urbdrc/client/data_transfer.c [hunk 2]
```diff
@@ -2319,12 +2319,9 @@ void* urbdrc_process_udev_data_transfer(void* arg)
 	pdev = udevman->get_udevice_by_UsbDevice(udevman, UsbDevice);
 	if (pdev == NULL || pdev->isSigToEnd(pdev))
 	{
-		if (transfer_data)
-		{
-			if (transfer_data->pBuffer)
-				zfree(transfer_data->pBuffer);
-			zfree(transfer_data);
-		}
+		if (transfer_data->pBuffer)
```
Comment (akallabeth) [inline channels/urbdrc/client/data_transfer.c]: that looks more like an issue with cppcheck.
There is no way to know the function will always be called with a non NULL argument (although it currently is the case)

Comment (chipitsine) [inline channels/urbdrc/client/data_transfer.c]: it is not cppcheck, I checked by eyes.
cppcheck is correct here

Comment (akallabeth) [inline channels/urbdrc/client/data_transfer.c]: so, transfer_data is checked somewhere else? As already mentioned, that may be the case with current call graph but you can not count on that. (Ok, just expanded the start of the function view, the whole argument checking here is broken anyway :/)

File: client/X11/xf_rail.c (hunk 1 above)
Comment (akallabeth) [inline client/X11/xf_rail.c]: add argument checks in ```xf_UpdateWindowArea``` before any argument is used and abort if they are NULL or similar

Comment (chipitsine) [inline client/X11/xf_rail.c]: we can annotate it like

void* urbdrc_process_udev_data_transfer(void* arg) __attribute__((nonnull(1)));

and involve static analyzer.

or, yes, we can check it against NULL

**Comment by akallabeth** [comment]
we only have the null check as option, annotations are too compiler specific :/

**Comment by chipitsine** [comment]
well, there's jenkins, it might run clang static analyzer, which pays attention to annotations.
ok, I'll add check against NULL

**Comment by akallabeth** [comment]
@chipitsine If you add the argument check in ```xf_UpdateWindowArea``` I'll merge this one.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@chihchun Ok, looks like you have a typo in your latest commit.

**Comment by akallabeth** [comment]
@chihchun would be nice if you could squash the last 2 commits.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Comment by akallabeth** [comment]
@freerdp-bot now come on, test!

**Comment by akallabeth** [comment]
@freerdp-bot this is not your day, hmm? test please!

**Comment by akallabeth** [comment]
@freerdp-bot oh, this gets old! test

**Comment by chihchun** [comment]
@akallabeth I pretty sure you tagged the wrong person.

**Comment by akallabeth** [comment]
@chihchun Oops, sorry :) Never trust autocomplete

**Pull request merged**